edgexfoundry/edgex-examples
# feat: support usb cameras in camera mgmt app

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-examples/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:


## Testing Instructions


- Note that the port for EVAM result streams has changed from `8554` to `8555` to avoid conflicts with device-usb-camera service.

Steps  to configure usb device service:

- clone egdex-compose repo
- checkout levski branch
- cd into `compose-builder` folder
- run `make gen no-secty ds-usb-camera ds-onvif-camera`
- edit `docker-compose.yml`
- find `device-usb-camera` service
- under `environment`, add `DRIVER_RTSPSERVERHOSTNAME: <your-local-ip-address>`, inserting your local ip address.
- under `ports`, find the one for port `8554` and change the `host_ip` from `127.0.0.1` to either `0.0.0.0` or the ip address you put in the previous step.
- save the file
- run `make up`
Note: re-running `make gen` or `make run` may override your changes.
